### ISSUE TITLE
App Crashes in in Library Activity

### ISSUE THREAD

**Issue by siddharth2010** [kiwix/kiwix-android#975]
**Describe the bug**

When the zim file is downloading, opening the Downloading Fragement using the Notifications, followed by clicking the online tab, leads to a crash.

**Steps to reproduce the behavior:**
1. Start Downloading a zim file
2. Open Notifications and tap on the Downloading notification, to open the downloading fragment
3. Click on the online tab
4. The app crashes

**Screenshots**

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/32238400/52594099-aae72280-2e70-11e9-9480-6a0e0125380c.gif)

Also sometimes, the zim Downloading notification is still left on the screen:

![screenshot_20190212-024642](https://user-images.githubusercontent.com/32238400/52594163-cf42ff00-2e70-11e9-9513-e0d1276bd281.png)

**Environment**
- Version of Kiwix Android : 2.4 Build 55
- Device : Samsung S6
- OS version : Android 7.0

**Comment by siddharth2010** [comment]
Here is the Log for the crash:
```
2019-02-12 02:54:23.443 3093-3093/org.kiwix.kiwixmobile E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.kiwix.kiwixmobile, PID: 3093
    java.lang.RuntimeException: Unable to stop activity {org.kiwix.kiwixmobile/org.kiwix.kiwixmobile.zim_manager.ZimManageActivity}: java.lang.IllegalArgumentException: Receiver not registered: org.kiwix.kiwixmobile.zim_manager.library_view.LibraryFragment$NetworkBroadcastReceiver@e7e9acc
        at android.app.ActivityThread.performStopActivityInner(ActivityThread.java:4241)
        at android.app.ActivityThread.handleStopActivity(ActivityThread.java:4290)
        at android.app.ActivityThread.-wrap28(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1665)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6682)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1520)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1410)
     Caused by: java.lang.IllegalArgumentException: Receiver not registered: org.kiwix.kiwixmobile.zim_manager.library_view.LibraryFragment$NetworkBroadcastReceiver@e7e9acc
        at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:1054)
        at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1359)
        at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:651)
        at org.kiwix.kiwixmobile.zim_manager.library_view.LibraryFragment.onStop(LibraryFragment.java:141)
        at androidx.fragment.app.Fragment.performStop(Fragment.java:2662)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManager.java:1519)
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1784)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManager.java:1852)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManager.java:3269)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStop(FragmentManager.java:3250)
        at androidx.fragment.app.FragmentController.dispatchStop(FragmentController.java:245)
        at androidx.fragment.app.FragmentActivity.onStop(FragmentActivity.java:641)
        at androidx.appcompat.app.AppCompatActivity.onStop(AppCompatActivity.java:184)
        at android.app.Instrumentation.callActivityOnStop(Instrumentation.java:1297)
        at android.app.Activity.performStop(Activity.java:7168)
        at android.app.ActivityThread.performStopActivityInner(ActivityThread.java:4238)
        at android.app.ActivityThread.handleStopActivity(ActivityThread.java:4290) 
        at android.app.ActivityThread.-wrap28(ActivityThread.java) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1665) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:154) 
        at android.app.ActivityThread.main(ActivityThread.java:6682) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1520) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1410)
```

**Comment by siddharth2010** [comment]
Also, I noticed that this also happens in case the device is rotated when the zim file is downloading.

**To Reproduce another crash scenario:**
1. Start downloading a zim file
2. Go to the downloading tab
3. Rotate the device
4. Stop the download
5. Go to another tab
6. The app crashes

Here is the gif:
![ezgif com-video-to-gif 1](https://user-images.githubusercontent.com/32238400/52595809-c522ff80-2e74-11e9-80d2-4a299e4f2725.gif)

**Comment by siddharth2010** [comment]
Here is the log for the second scenario:

```
2019-02-12 03:19:47.520 8013-8013/org.kiwix.kiwixmobile E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.kiwix.kiwixmobile, PID: 8013
    java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.ListView.invalidateViews()' on a null object reference
        at org.kiwix.kiwixmobile.downloader.DownloadService.pauseDownload(DownloadService.java:283)
        at org.kiwix.kiwixmobile.downloader.DownloadFragment$DownloadAdapter.setPlayState(DownloadFragment.java:255)
        at org.kiwix.kiwixmobile.downloader.DownloadFragment$DownloadAdapter.lambda$getView$5(DownloadFragment.java:324)
        at org.kiwix.kiwixmobile.downloader.-$$Lambda$DownloadFragment$DownloadAdapter$LxyhzTeoe7ZUFXuWnasr5s63_Bc.onClick(lambda)
        at android.view.View.performClick(View.java:6205)
        at android.view.View$PerformClick.run(View.java:23653)
        at android.os.Handler.handleCallback(Handler.java:751)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6682)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1520)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1410)
```

**Comment by siddharth2010** [comment]
@mhutti1 @abdulwd could you please see if you get the crash on your end too.

**Comment by mhutti1** [comment]
There are lots of known download crashes so this wouldn't surprise me.

**Comment by ChandanNaik999** [comment]
@siddharth2010  for me the app crashes as soon as I stop downloading the app in Landscape mode

The error is: 
2019-04-08 03:49:28.550 14337-14398/org.kiwix.kiwixmobile E/kiwix: Exception reading article A/Main_Page.html from zim file
    java.io.IOException: write failed: EPIPE (Broken pipe)
        at libcore.io.IoBridge.write(IoBridge.java:501)
        at java.io.FileOutputStream.write(FileOutputStream.java:316)
        at org.kiwix.kiwixmobile.ZimContentProvider$TransferThread.run(ZimContentProvider.java:487)
     Caused by: android.system.ErrnoException: write failed: EPIPE (Broken pipe)
        at libcore.io.Posix.writeBytes(Native Method)
        at libcore.io.Posix.write(Posix.java:273)
        at libcore.io.BlockGuardOs.write(BlockGuardOs.java:319)
        at libcore.io.IoBridge.write(IoBridge.java:496)
        at java.io.FileOutputStream.write(FileOutputStream.java:316) 
        at org.kiwix.kiwixmobile.ZimContentProvider$TransferThread.run(ZimContentProvider.java:487)

**Comment by CodetoInvent** [comment]
Hello I am Abdul Malik @codetoinvent ,
I have fixed the above bug you have mention in PR #1161 and #1162 
please check it out @siddharth2010 @mhutti1 and Accept my PR
Thanks

**Comment by macgills** [comment]
Fixed in #1170
